homeassistant/config_entries.py

```diff
@@ -266,13 +268,14 @@ class ConfigFlowResult(FlowResult, total=False):
     version: int
 
 
-class ConfigEntry:
+class ConfigEntry(Generic[_DataT]):
     """Hold a configuration entry."""
 
     entry_id: str
     domain: str
     title: str
     data: MappingProxyType[str, Any]
+    shared_data: _DataT
```

Comment:
I think I prefer `runtime_data` as a term over `shared_data` (but that is a personal preference).